gadget/install/install.go

```diff
@@ -96,27 +96,25 @@ func saveStorageTraits(mod gadget.Model, allLaidOutVols map[string]*gadget.LaidO
 	return nil
 }
 
-func maybeEncryptPartition(odls *onDiskAndLaidoutStructure, encryptionType secboot.EncryptionType, sectorSize quantity.Size, perfTimings timings.Measurer) (fsParams *mkfsParams, encryptionKey keys.EncryptionKey, err error) {
+func maybeEncryptPartition(laidOut *gadget.LaidOutStructure, encryptionType secboot.EncryptionType, sectorSize quantity.Size, perfTimings timings.Measurer) (fsParams *mkfsParams, encryptionKey keys.EncryptionKey, err error) {
 	mustEncrypt := (encryptionType != secboot.EncryptionTypeNone)
-	onDisk := odls.onDisk
-	laidOut := odls.laidOut
 	// fsParams.Device is the kernel device that carries the
 	// filesystem, which is either the raw /dev/<partition>, or
 	// the mapped LUKS device if the structure is encrypted (if
 	// the latter, it will be filled below in this function).
 	fsParams = &mkfsParams{
-		// Filesystem and label are as specified in the gadget
-		Type:  laidOut.Filesystem,
-		Label: laidOut.Label,
-		// Rest come from disk data
-		Device:     onDisk.Node,
-		Size:       onDisk.Size,
+		// fs is taken from gadget, as on disk one might be displayed as
+		// crypto_LUKS, which is not useful for formatting.
+		Type:       laidOut.GadgetStructure.Filesystem,
+		Label:      laidOut.Label,
```

Comment:
is label correct here?

Reply:
I mean, is not clear why it is not laidout.GadgetStructure.Label ?

Reply:
I made this debug change on master:

```
diff --git a/gadget/install/content.go b/gadget/install/content.go
index 6bd32c599f..313a6de959 100644
--- a/gadget/install/content.go
+++ b/gadget/install/content.go
@@ -55,6 +55,7 @@ type onDiskAndLaidoutStructure struct {
 // that sector size is used when creating the filesystem, otherwise if it is
 // zero, automatic values are used instead.
 func makeFilesystem(params mkfsParams) error {
+       fmt.Println(params.Device, params.Type, params.Label)
        logger.Debugf("create %s filesystem on %s with label %q", params.Type, params.Device, params.Label)
        if err := mkfsImpl(params.Type, params.Device, params.Label, params.Size, params.SectorSize); err != nil {
                return err
```

and a similar one here, and running the test very different things get printed in some cases. Maybe master is wrong in the tests? Either way, it seems we should add some assertions in master first so we are more confident making these changes